packages/admin-panel/src/dataTables/useDataTable.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+const useParameters = ({ config, onEditField }) => {
+  const { additionalParameters = [] } = config;
+
+  const onAdd = () => {
+    const defaultNewParameter = {
+      id: `parameter_${additionalParameters.length}`,
+    };
+
+    onEditField('config', {
+      ...config,
+      additionalParameters: [...additionalParameters, defaultNewParameter],
+    });
+  };
+
+  const onDelete = selectedParameterId => {
+    const newParameterList = additionalParameters.filter(p => p.id !== selectedParameterId);
+    onEditField('config', { ...config, additionalParameters: newParameterList });
+  };
+
+  const onChange = (id, key, newValue) => {
+    const newParameters = [...additionalParameters];
+    const index = additionalParameters.findIndex(p => p.id === id);
+
+    // validate name input
+    if (key === 'name') {
+      try {
+        if (newValue === '') {
+          throw new Error('Cannot be empty');
+        }
+
+        if (additionalParameters.findIndex(p => p.name === newValue) !== -1) {
+          throw new Error('Duplicated parameter name');
+        }
+
+        const regex = new RegExp('^[A-Za-z0-9_]+$');
```

Comment:
Is this because we want the params to be used in sql and so they have to be simple no-space words? 

If so can you leave a comment here explaining why we have this restriction.